dbfit-java/core/src/main/java/dbfit/util/sql/PreparedStatements.java

```diff
@@ -8,11 +8,23 @@ private static String buildParamList(int numberOfParameters) {
         return join(repeat("?", numberOfParameters), ", ");
     }
 
-    public static String buildStoredProcedureCall(String procName, int numberOfPararameters) {
-        return "{ call " + procName + "(" + buildParamList(numberOfPararameters) + ")}";
+    private static String returnValueParam(boolean returnsValue) {
+        return returnsValue ? "? = " : "";
     }
 
-    public static String buildFunctionCall(String procName, int numberOfParameters) {
-        return "{ ? = call " + procName + "(" + buildParamList(numberOfParameters - 1) + ")}";
+    private static String storedRoutineExecution(String routineName, int numberOfParameters) {
+        return "call " + routineName + "(" + buildParamList(numberOfParameters) + ")";
+    }
+
+    public static String storedRoutineStatement(
+            String routineName, int numberOfParams, boolean hasReturnValueParam) {
+        int executionParams = numberOfParams - (hasReturnValueParam ? 1 : 0);
+        return returnValueParam(hasReturnValueParam) +
+            storedRoutineExecution(routineName, executionParams);
+    }
+
+    public static String storedRoutineCall(
+            String routineName, int numberOfParams, boolean hasReturnValueParam) {
+        return "{ " + storedRoutineStatement(routineName, numberOfParams, hasReturnValueParam) + " }";
```

Comment:
We ended up with several names: `execution`, `statement`, `call`... I may add `invocation`. I'm wondering whether we can reduce the confusion, maybe there is no ideal answer... General ideas:
* The public function seems most eligible to tag with `statement` as part of its name (the others are some fractions of the final statement)
* I'm not sure whether all slicing into separate functions helps (at the moment). An example with less procs might look like:

``` java 
    public static String buildStoredRoutineCallText(
            String name, int numberOfParams, boolean hasReturnValueParam) {
        int numArgs = numberOfParams - (hasReturnValueParam ? 1 : 0);
        String resultAssignment = hasReturnValueParam ? "? = " : "";
        String voidInvocation = name + "(" + buildParamList(numArgs) + ")";
        return "{ " + resultAssignment + "call " + voidInvocation + " }";
    }
```